src/core/interfaces.d.ts

```diff
@@ -124,6 +124,23 @@ export interface Theme {
 	[key: string]: object;
 }
 
+export interface Variant {
+	root: string;
+}
+
+export interface ThemeVariant {
```

Comment:
Would this be better as `ThemeWithVariant` and then the interface below `ThemeWithVariants`?